libavformat/hls.c

```diff
@@ -226,12 +226,24 @@ typedef struct HLSContext {
     int variant_count;
 } HLSContext;
 
+// If both index and bandwidth are selected, the ensure both match
 static int is_variant_selected(HLSContext* c, const char* current_bandwidth) {
     // If both selected bandwidth and selected variant are default, then add this program
     if (!c->selected_bandwidth && c->selected_variant_index == -1) {
         return 1;
     }
     // If a variant is selected and we've counted enough variants
+    // and the bandwidth is selected
+    else if (c->selected_variant_index != -1 && c->selected_bandwidth &&
```

Comment:
if none of them are selected, do we return all variants?

Reply:
I think we do .. it's the first condition in the function!

Reply:
We do. This way we can do filtering in the application if we want in the future.